test/gtest/ucp/test_ucp_sockaddr.cc

```diff
@@ -2451,28 +2483,32 @@ class test_ucp_sockaddr_protocols : public test_ucp_sockaddr {
             std::string rb(size, 'y');
             std::string shdr(hdr_size, 'x');
             std::string rhdr(hdr_size, 'y');
+            ucp_mem_h smemh(NULL);
             ucp_mem_h rmemh(NULL);
 
-            if (recv_prereg_memh) {
-                rmemh = receiver().mem_map(&rb[0], size);
+            if (prereg_memh_flags & SEND_PREREG_MEMH) {
+                smemh = sender().mem_map(&sb[0], size);
             }
-            
-            if (recv_mds != NULL) {
-                ucp_tl_md_t *tl_md;
-                ucs_carray_for_each(tl_md, receiver().ucph()->tl_mds,
-                                    receiver().ucph()->num_mds) {
-                    recv_mds->push_back(tl_md->md);
-                    tl_md->md = NULL;
-                }
+
+            if (prereg_memh_flags & RECV_PREREG_MEMH) {
+                rmemh = receiver().mem_map(&rb[0], size);
             }
 
+            save_and_reset_mds(sender(), send_mds);
```

Comment:
1. maybe not check send_mds==NULL - instead check prereg flag to decide if need to save and reset the MDs.
2. then, the vector does not need to be passed as a parameter - it could be a local variable in test_am_send_recv function